arch/xtensa/src/esp32/esp32_spiflash.c

```diff
@@ -1343,10 +1343,6 @@ FAR struct mtd_dev_s *esp32_spiflash_alloc_mtdpart(void)
   uint32_t startblock;
   uint32_t size;
 
-  /* Initiliaze the mutex */
-
-  nxsem_init(&priv->exclsem, 0, 1);
```

Comment:
Strange, I didn't find where are you calling nxsem_init(&s_exclsem, 0, 1) now?

Reply:
I use global variable `static sem_t g_exclsem = SEM_INITIALIZER(1);` to initialize the semaphore, and all MTDs use the one semaphore, so that they will not operate the SPI at the same time.